<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DataAggregationExporter_Widget</class>
 <widget class="QWidget" name="DataAggregationExporter_Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Data Aggregation Exporter</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="label_description">
     <property name="text">
      <string>Export aggregated data across time intervals. Select a Digital Interval Series as the basis for aggregation, then choose data keys and transformations to export.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   
   <!-- Interval Source Selection -->
   <item>
    <layout class="QHBoxLayout" name="interval_source_layout">
     <item>
      <widget class="QLabel" name="label_interval_source">
       <property name="text">
        <string>Interval Source:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="interval_source_combo">
       <property name="toolTip">
        <string>Select the Digital Interval Series to use as the basis for aggregation</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="interval_source_spacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>

   <!-- Available Data Table -->
   <item>
    <widget class="QLabel" name="label_available_data">
     <property name="text">
      <string>Available Data Keys:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTableWidget" name="available_data_table">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
     </property>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
    </widget>
   </item>

   <!-- Transformation Selection -->
   <item>
    <widget class="QGroupBox" name="transformation_group">
     <property name="title">
      <string>Add Export Column</string>
     </property>
     <layout class="QGridLayout" name="transformation_layout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_transformation">
        <property name="text">
         <string>Transformation:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="transformation_combo">
        <property name="toolTip">
         <string>Select the transformation to apply to the selected data</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_column_name">
        <property name="text">
         <string>Column Name:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="column_name_edit">
        <property name="toolTip">
         <string>Custom name for this column in the exported CSV</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QPushButton" name="add_export_button">
        <property name="text">
         <string>Add to Export List</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>

   <!-- Export List Table -->
   <item>
    <widget class="QLabel" name="label_export_list">
     <property name="text">
      <string>Export List:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTableWidget" name="export_list_table">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
     </property>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
    </widget>
   </item>
   
   <item>
    <layout class="QHBoxLayout" name="export_list_buttons_layout">
     <item>
      <widget class="QPushButton" name="remove_export_button">
       <property name="text">
        <string>Remove Selected</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="clear_export_button">
       <property name="text">
        <string>Clear All</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="export_buttons_spacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>

   <!-- CSV Export Options -->
   <item>
    <widget class="QGroupBox" name="csv_options_group">
     <property name="title">
      <string>CSV Export Options</string>
     </property>
     <layout class="QGridLayout" name="csv_options_layout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_delimiter">
        <property name="text">
         <string>Delimiter:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="delimiter_combo">
        <item>
         <property name="text">
          <string>Comma</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Space</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Tab</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_line_ending">
        <property name="text">
         <string>Line Ending:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="line_ending_combo">
        <item>
         <property name="text">
          <string>LF (\n)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>CRLF (\r\n)</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QCheckBox" name="save_header_checkbox">
        <property name="text">
         <string>Include Header Row</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_precision">
        <property name="text">
         <string>Decimal Precision:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="precision_spinbox">
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>10</number>
        </property>
        <property name="value">
         <number>3</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>

   <!-- Export Button -->
   <item>
    <widget class="QPushButton" name="export_csv_button">
     <property name="text">
      <string>Export to CSV...</string>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui> 